powerplantmatching/cleaning.py

```diff
@@ -449,7 +450,10 @@ def aggregate_units(
 
     df = cliques(df, duplicates)
     df = df.groupby("grouped").agg(props_for_groups)
-    df[str_cols] = df[str_cols].replace("", np.nan)
+
+    # Downcasting in replace is deprecated
+    with pd.option_context("future.no_silent_downcasting", True):
+        df[str_cols] = df[str_cols].replace("", np.nan).infer_objects(copy=False)
```

Comment:
```suggestion
    df[str_cols] = df[str_cols].astype('object').replace("", np.nan)
```
this could already do it...

Reply:
It does not, but 
```df[str_cols] = df[str_cols].replace("", pd.NA)```
does. 
I am not fully sure, but I think the issue is not the incoming dtype but the different types which replace casts to, since `np.nan` is float based.

`pd.NA` is made for this case, but it is still [experimental and has a slightly different behaviour](https://pandas.pydata.org/docs/user_guide/missing_data.html#na-semantics) for equality checks. But I don't think that's going to cause any problems here, and it would be good to start using it anyway, because we're no longer mixing types.